plugins/spidmx/README.md

```diff
@@ -0,0 +1,46 @@
+Native SPI DMX Plugin
+=====================
+
+This plugin samples the Serial Peripheral Interface (SPI) at 2MHz to
+receive DMX data. You need to connect only the *MISO* pin to the *receive*
+pin of a transceiver chip like the *MAX485* or *SN75176*.
+
+Repeatedly, blocks of 4096 bytes are read as a batch operation and parsed
+afterwards. We must hope that a DMX packet is contained in a block as a
+whole. Since often a packet will be cut off by the block end, the higher
+channels are updated less frequently than the lower ones. However, you can
+set a larger block size if you want a higher refresh rate for the high
+channels (which results in more stuttering in all channels, as the time
+between two SPI read operation increases).
+
+
+## Raspberry Pi configuration
+
+On the Raspberry Pi, you need to enable SPI and increase the buffer size:
+
+* Run `sudo raspi-config` and enable SPI.
+* Add `spidev.bufsiz=65536` to `/etc/cmdline.txt` (in the same line). This
+  is the maximum value. The value specified must be at least the block
+  length set below. The default `spidev.bufsiz` is 4096.
+* Reboot.
+* Check `cat /sys/module/spidev/parameters/bufsiz` that it now returns the
+  new value.
+
+
+## Config file: `ola-spidmx.conf`
+
+`enabled = true`  
+Enable this plugin (DISABLED by default).
+
+`device_prefix = <string>`  
+The prefix of files to match in `/dev`. Usually set to `spidev`. Each match
+will instantiate a device.
+
+### Per Device Settings
+
+**Note:** Substitute `<device>` with the full device name, e.g.
+`/dev/spidev0.0`.
+
+`<device>-blocklength = 4096`  
+How many SPI bytes (= DMX bits) should be received (optional). The default
+is 4096, but 8192 is recommended.
```

Comment:
Out of interest, how come we don't default to the value we recommend?

Reply:
On the Raspberry Pi, without further configuration, a 4kB read / trabsmit buffer is available. If one wants to read more than that in a single call, the buffer size needs to be increased via a command line call.

Reply:
Ah yes, a few lines up the same file. I wonder if it's worth an N.B. here when recommending 8192 that they'll need to do some local config to enable this.

Reply:
The call used to read / write ([ioctl](http://man7.org/linux/man-pages/man2/ioctl.2.html), in [SPIDMXWidget.cpp](https://github.com/FloEdelmann/ola/blob/b8ecf02095b3bff6a9c5e0a3ebdda374000394b6/plugins/spidmx/SPIDMXWidget.cpp#L112)) does not return the number of received / transmitted bytes but only an unspecified non-negative integer or -1 in an error case.

Since the call fails if the buffer is not big enough, the check should already be enough, I guess.

Reply:
That's a shame. Perhaps the error on the line below needs a hint about buffer sizes though, if changing the value to 4097 will cause it to break?

Reply:
Like this?